src/coreclr/jit/optimizer.cpp

```diff
@@ -718,7 +718,7 @@ bool Compiler::optPopulateInitInfo(unsigned loopInd, GenTree* init, unsigned ite
 
     // RHS can be constant or local var.
     // TODO-CQ: CLONE: Add arr length for descending loops.
-    if (rhs->gtOper == GT_CNS_INT && rhs->TypeGet() == TYP_INT)
+    if (rhs->gtOper == GT_CNS_INT && ((rhs->TypeGet() == TYP_INT) || (rhs->TypeGet() == TYP_I_IMPL)))
```

Comment:
This works by allowing either `TYP_INT` or `TYP_I_IMPL`
In the case of `TYP_I_IMPL` it also looks for `GT_CAST` to that type and gets the underlying node when that exists